test/fast/Sourcing nvm.sh should not modify parameters of caller

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+set -- yes
```

Comment:
can you elaborate on why you're using `set` here instead of `. ../../nvm.sh yes`?